website_url_sitemap_exclusion/models/website.py

```diff
@@ -0,0 +1,29 @@
+#  Copyright 2023 Simone Rubino - TAKOBI
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import re
+from odoo import api, fields, models
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    forbidden_url_ids = fields.One2many(
+        comodel_name="website.forbidden.url",
+        inverse_name="website_id",
+        string="Forbidden URLs regex",
+    )
+
+    @api.multi
+    def enumerate_pages(self, query_string=None, force=False):
+        pages = super().enumerate_pages(query_string=query_string, force=force)
+        forbidden_url_regexes = self.mapped("forbidden_url_ids.regex")
+        forbidden_url_compiled_regexes = list(map(re.compile, forbidden_url_regexes))
+        for page in pages:
+            page_url = page["loc"]
+            for forbidden_url_compiled_regex in forbidden_url_compiled_regexes:
+                is_url_forbidden = forbidden_url_compiled_regex.match(page_url)
+                if is_url_forbidden:
+                    break
+            else:
+                yield page
```

Comment:
```suggestion
            if not self._enumerate_pages_forbidden(page):
                yield page

    def _enumerate_pages_forbidden(self, page):
        page_url = page["loc"]
        for forbidden_url_compiled_regex in forbidden_url_compiled_regexes:
            if forbidden_url_compiled_regex.match(page_url):
                return True
```

Reply:
I am not really a fan of `return`ing in a loop so I have done something similar to what you suggested, please have a look at the new changes.